src/mpm/mpm.h

```diff
@@ -428,9 +428,14 @@ class MPElement : public Element {
         } else {
             bNodes = this->giveBoundaryEdgeNodes(ibc);
         }
+        int num = 0;
```

Comment:
In my opinion, the better approach here would be to have just a single pass over bNodes with giveUnknownVector call done only once (this is rather costly).
I would suggest to use std::list<double> to collect entries and  then convert it to answer, with size known at this stage.

Reply:
Fixed with `std::list<double>` now.